Celeste.Mod.mm/Mod/Everest/Everest.Content.cs

```diff
@@ -725,108 +725,316 @@ public static void Add(string path, ModAsset metadata)
             public static string GuessType(string file, out Type type, out string format) {
                 type = typeof(object);
                 format = Path.GetExtension(file) ?? "";
-                if (format.Length >= 1)
-                    format = format.Substring(1);
 
-                // Assign game asset types
-                if (format == "dll") {
+                if (format.Length < 1)
+                    return file;
+
+                format = format[1..];
+
+                ReadOnlySpan<char> fileSpan = file.AsSpan();
+                int fileSeparator = fileSpan.LastIndexOf('/') + 1;
+
+                // folder with / at the end
+                ReadOnlySpan<char> directorySpan = fileSpan[..fileSeparator]; // don't use Path.GetDirectoryName as it replaces '/' with '\' :catplush:
+                // file
+                ReadOnlySpan<char> fileNameSpan = fileSpan[fileSeparator..];
+                // file without extension
+                ReadOnlySpan<char> fileNameOnlySpan = Path.GetFileNameWithoutExtension(fileNameSpan);
+
+                bool shouldSendWarning = true;
```

Comment:
this varname is confusing; i'd recommend something like `alreadyWarned` & invert the behaviour

Reply:
Fair - also I noticed that the warning only logs the file name instead of its path, which is not the best.
My biggest gripe is that I can't access the information telling me what mod the file path belongs to.

Reply:
hmm yea, having the path would be nice. i wouldn't say it's a dealbreaker but would be good to look into